catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RegistryPublicationServiceImpl.java

```diff
@@ -142,7 +142,11 @@ public void update(Metacard metacard) throws FederationAdminException {
                 .collect(Collectors.toCollection(HashSet::new));
 
         if (CollectionUtils.isNotEmpty(locations)) {
-            federationAdminService.updateRegistryEntry(metacard, locations);
+            try {
+                federationAdminService.updateRegistryEntry(metacard, locations);
+            } catch (FederationAdminException e){
+                federationAdminService.addRegistryEntry(metacard, locations);
```

Comment:
The only reason an exception would occur above is if underlying records weren't present to be updated? No other cause where that exception might be thrown and a call to `addRegistryEntry` would be in error?

And, if it is the case that a failed updated should automatically lead to an add, is there a way to manage that process flow without using exception processing? Using exceptions for flow control is a code smell.